### PR TITLE
docs: document service_name label in upgrade guide

### PR DESCRIPTION
**Pull request by slim-bean**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Review by sandeepsukhani — APPROVED** [review]

File: docs/sources/setup/upgrade/_index.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ Here is the shortlist of things we think most people may encounter:
   * The `shared_store` config is removed. Refer to [Removed `shared_store` and `shared_store_key_prefix` from shipper configuration](#removed-shared_store-and-shared_store_key_prefix-from-shipper-configuration).
   * Loki now enforces a max line size of 256KB by default (you can disable this or increase this but this is how Grafana Labs runs Loki). Refer to [Changes to default configure values](#changes-to-default-configuration-values-in-30).
   * Loki now enforces a max label limit of 15 labels per series, down from 30. Extra labels inflate the size of the index and reduce performance, you should almost never need more than 15 labels. Refer to [Changes to default configure values](#changes-to-default-configuration-values-in-30).
+  * Loki will automatically attempt to populate a `service_name` label on ingestion. Refer to [`service_name` label](#service_name-label)
```
Comment (JStickler) [inline docs/sources/setup/upgrade/_index.md:55]:
```suggestion
  * Loki will automatically attempt to populate a `service_name` label on ingestion. Refer to [`service_name` label](#service_name-label).
```

File: docs/sources/setup/upgrade/_index.md [hunk 2]
```diff
@@ -82,6 +83,29 @@ Structured Metadata is enabled by default in Loki 3.0, however, it requires your
...
+  - app_kubernetes_io_name
+  - container
+  - container_name
+  - component
+  - workload
+  - job
+
+If no label is found matching the list, a value of `unknown_service` is applied.
+
+You can change this list by providing a list of labels to the `discover_service_name` in the [limits_config](/docs/loki/<LOKI_VERSION>/configure/#limits_config) block.
```
Comment (JStickler) [inline docs/sources/setup/upgrade/_index.md:105]:
```suggestion
You can change this list by providing a list of labels to `discover_service_name` in the [limits_config](/docs/loki/<LOKI_VERSION>/configure/#limits_config) block.
```

File: docs/sources/setup/upgrade/_index.md [hunk 3]
```diff
@@ -82,6 +83,29 @@ Structured Metadata is enabled by default in Loki 3.0, however, it requires your
...
+
+**You can disable this by providing an empty value for `discover_service_name`**
```
Comment (JStickler) [inline docs/sources/setup/upgrade/_index.md:107]:
```suggestion
**You can disable this by providing an empty value for `discover_service_name`.**
```

**Review by JStickler — COMMENTED** [review]
[Docs team]  LGTM other than a few missing periods.

**Pull request merged**